storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -3600,7 +3600,7 @@ public ComponentPageInfo getComponentPageInfo(String topoId, String componentId,
                 List<Integer> tasks = compToTasks.get(StormCommon.EVENTLOGGER_COMPONENT_ID);
                 tasks.sort(null);
                 // Find the task the events from this component route to.
-                int taskIndex = TupleUtils.listHashCode(Arrays.asList(componentId)) %
+                int taskIndex = (TupleUtils.listHashCode(Arrays.asList(componentId)) % tasks.size() + tasks.size()) %
```

Comment:
@arunmahadevan Thanks for pointing this out, I will send another pull request with the change.